textdb/textdb-perftest/src/main/java/edu/uci/ics/textdb/perftest/sample/SampleExtraction.java

```diff
@@ -0,0 +1,161 @@
+import edu.uci.ics.textdb.dataflow.regexmatch.RegexMatcher;
+import edu.uci.ics.textdb.dataflow.sink.FileSink;
+import edu.uci.ics.textdb.dataflow.sink.IndexSink;
+import edu.uci.ics.textdb.dataflow.source.TupleStreamSourceOperator;
+import edu.uci.ics.textdb.engine.Engine;
+import edu.uci.ics.textdb.perftest.promed.PromedSchema;
+import edu.uci.ics.textdb.perftest.utils.PerfTestUtils;
+import edu.uci.ics.textdb.storage.DataStore;
+import edu.uci.ics.textdb.storage.writer.DataWriter;
+
+public class SampleExtraction {
+        
+    public static final String promedFilesDirectory = "./sample-data-files/promed/";
+    public static final String promedIndexDirectory = "./index/standard/promed/"; 
+    public static final DataStore promedDataStore = new DataStore(promedIndexDirectory, PromedSchema.PROMED_SCHEMA);
+    
+    public static void main(String[] args) throws Exception {
+        
+//         writeSampleIndex();
+        
+        extractPersonLocation();
+    }
+    
+    public static ITuple parsePromedHTML(String fileName, String content) {
+        try {
+            Document parsedDocument = Jsoup.parse(content);
+            String mainText = parsedDocument.getElementById("preview").text();
+            ITuple tuple = new DataTuple(PromedSchema.PROMED_SCHEMA, new StringField(fileName), new TextField(mainText));
+            return tuple;
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    
+    public static void writeSampleIndex() throws Exception {
+        // clear the directory
+        new DataWriter(promedDataStore, LuceneAnalyzerConstants.getStandardAnalyzer()).clearData();
+        
+        File sourceFileFolder = new File(promedFilesDirectory);
+        ArrayList<ITuple> fileTuples = new ArrayList<>();
+        for (File htmlFile : sourceFileFolder.listFiles()) {
+            StringBuilder sb = new StringBuilder();
+            Scanner scanner = new Scanner(htmlFile);
+            while (scanner.hasNext()) {
+                sb.append(scanner.nextLine());
+            }
+            scanner.close();
+            ITuple tuple = parsePromedHTML(htmlFile.getName(), sb.toString());
+            if (tuple != null) {
+                fileTuples.add(tuple);
+            }
+        }
+
+        ISourceOperator fileSource = new TupleStreamSourceOperator(fileTuples,
+                PromedSchema.PROMED_SCHEMA);
+
+        IndexSink standardIndexSink = new IndexSink(promedIndexDirectory, PromedSchema.PROMED_SCHEMA,
+                new StandardAnalyzer(), false);
+        standardIndexSink.setInputOperator(fileSource);
+        Plan standardIndexPlan = new Plan(standardIndexSink);
+
+        Engine engine = Engine.getEngine();
+        engine.evaluate(standardIndexPlan);
+
+        Engine.getEngine().evaluate(standardIndexPlan);
+    }
+
+    /*
+     * This is the DAG of this extraction plan.
+     * 
+     *                        --> Regex (a ... man) --
```

Comment:
Should the diagram also include the projection operators?

Added @qinhuah and @huizhang11 for their information.